From cd6d69a414711be7a66f676f685aab007592dbbc Mon Sep 17 00:00:00 2001
From: Paul Bakker <p.j.bakker@polarssl.org>
Date: Thu, 6 Feb 2014 15:43:21 +0100
Subject: [PATCH] Fixed testing with out-of-source builds using cmake

ChangeLog part stripped, because of collision, by hasufell@gentoo.org

---
 ChangeLog            |  1 +
 tests/CMakeLists.txt | 22 ++++++++++++++++++++++
 2 files changed, 23 insertions(+)

diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
index cf05bce..00e15e5 100644
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -83,3 +83,25 @@ add_test_suite(version)
 add_test_suite(xtea)
 add_test_suite(x509parse)
 add_test_suite(x509write)
+
+# Make data_files available in an out-of-source build
+if (NOT ${CMAKE_CURRENT_BINARY_DIR} STREQUAL ${CMAKE_CURRENT_SOURCE_DIR})
+    # Get OS dependent path to use in `execute_process`
+    file(TO_NATIVE_PATH "${CMAKE_CURRENT_BINARY_DIR}/data_files" link)
+    file(TO_NATIVE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/data_files" target)
+
+    if (UNIX)
+        set(command ln -s ${target} ${link})
+    else()
+        set(command cmd.exe /c mklink ${link} ${target})
+    endif()
+
+    execute_process(COMMAND ${command}
+                    RESULT_VARIABLE result
+                    ERROR_VARIABLE output)
+
+    if (NOT ${result} EQUAL 0)
+        message(FATAL_ERROR "Could not create symbolic link for: ${target} --> ${output}")
+    endif()
+endif()
+
-- 
1.8.5.5
